upstairs/src/lib.rs

```diff
@@ -892,20 +919,37 @@ async fn cmd_loop(
                  * TODO: Add a check here to make sure we are
                  * connected and in the proper state before we
                  * accept any commands.
+                 *
+                 * XXX Check the return code here and do something
+                 * about it.  If we fail in process_message, we should
+                 * handle it.
                  */
                 let _result =
                     process_message(&up_c, &m, up_coms_c.clone()).await;
+
+                if up_c.ds_deactivate(up_coms_c.client_id) {
+                    bail!("[{}] exits after deactivation", up_coms_c.client_id);
+                }
             }
-        });
-    }
+            Ok(())
+        })
+    };
 
+    tokio::pin!(pm_task);
     loop {
         tokio::select! {
             /*
-             * We set biased so the loop will always start with looking for
-             * an ACK from downstairs to avoid a backup.
+             * We set biased so the loop will:
+             * First make sure the pm task is running.
+             * Second, respond to any ACKs coming back from the downstairs.
```

Comment:
here's a comment I didn't update after moving things around - responding to ACKs doesn't occur here anymore

Reply:
Done in 9608a17